### PR TITLE
Add missing 'send' methods for uploading files

### PR DESCRIPTION
**Pull request by gharnly**

### CODE REVIEW

File: lib/ftpoison/base.ex [hunk 1]
```diff
@@ -37,33 +37,58 @@ defmodule FTPoison.Base do
...
+      @spec recv(PID, String.t(), String.t()) :: {:ok, PID}
       def recv(pid, remote_file, local_file) do
         case :ftp.recv(pid, to_char_list(remote_file), to_char_list(local_file)) do
           :ok -> pid
           e -> handle_error(e)
         end
       end
 
-      @spec start(String.t) :: {:ok, PID} | {:error, {atom, atom}}
+      @spec send(PID, String.t()) :: :ok | {:error, {atom, atom}}
```
Comment (scottzelley) [inline lib/ftpoison/base.ex]: are there any tests we could bring in?  I see the original functions have some basic testing.

Comment (gharnly) [reply lib/ftpoison/base.ex]: There are no tests for the recv methods which are used for downloading files. I added send methods for uploading files. Not sure how we would test these without actually communicating with an FTP site. We do host an FTP site for our Vendor SFTP project but I don't know that we want tests pushing files there, plus we'd have to figure out how to store encrypted credentials in this repo.

**Comment by gharnly** [comment]
Closing this as the "send" methods aren't working. I'm going to use a different hex package instead.

File: lib/ftpoison/base.ex (hunk 1 above)
Comment (calebshay) [reply lib/ftpoison/base.ex]: Also, the vendor sftp isn't ftp, it's sftp.  They have similar names, but the protocols are entirely different.

**Comment by calebshay** [comment]
@gharnly 
You probably wanted `send_bin`, not `send`, as FTP assumes ascii transfer by default and will try to make line ending changes to your data.
I'd be shocked if these don't actually work, they have been part of Erlang forever.  More likely, you are seeing issues with PORT vs PASV connections.

**Pull request closed without merging**